staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -395,14 +398,14 @@ func (p *patcher) applyPatch(_ context.Context, _, currentObject runtime.Object)
 	if err != nil {
 		return nil, err
 	} else if !currentObjectHasUID {
-		objToUpdate, patchErr = p.mechanism.createNewObject()
```

Comment:
I don't think this change is correct, we either want to call createNewObject or applyPatchToCurrentObject, here I think you're calling both